.github/scripts/stop-backend.sh

```
@@ -0,0 +1,116 @@
+
+# Kill process by PID file if it exists - kill entire process tree recursively
+KILLED_BY_PID=false
+if [ -f "$PID_FILE" ]; then
+  PID=$(cat "$PID_FILE")
+  if kill -0 "$PID" 2>/dev/null; then
+    echo "Stopping backend with PID $PID and entire process tree"
+
+    # Function to recursively get all descendants
+    get_descendants() {
+      local pid=$1
+      local descendants=$(pgrep -P "$pid" 2>/dev/null || true)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)